paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/SafeOreBreak.java

```diff
@@ -0,0 +1,199 @@
+import vg.civcraft.mc.civmodcore.players.settings.PlayerSetting;
+import vg.civcraft.mc.civmodcore.players.settings.PlayerSettingAPI;
+import vg.civcraft.mc.civmodcore.players.settings.gui.MenuSection;
+
+public final class SafeOreBreak extends BasicHack {
+
+	@AutoLoad
+	private List<String> ores;
+
+	private final List<BooleanOreSetting> oreSettings = new ArrayList<>();
+	private final List<List<Material>> breakOres = new ArrayList<>();
+
+	private final Set<Player> receivedWarning = Collections.newSetFromMap(new WeakHashMap<>());
+
+	public SafeOreBreak(final SimpleAdminHacks plugin, final BasicHackConfig config) {
+		super(plugin, config);
+	}
+
+	@Override
+	public void onEnable() {
+		super.onEnable();
+
+		MenuSection mainMenu = plugin.getSettingManager().getMainMenu();
+
+		OUTER:
+		for (String ore : ores) {
+			String[] parts = ore.split(",");
+			Material[] materialParts = new Material[parts.length];
+
+			for (int i = 0; i < parts.length; i++) {
+				String part = parts[i];
+				materialParts[i] = MaterialUtils.getMaterial(part);
+				if (materialParts[i] == null) {
+					logger.warning("Invalid material '"  + parts[0] + "'. Skipping.");
+					continue OUTER;
+				}
+			}
+
+			breakOres.add(Arrays.asList(materialParts));
+			String materialName = PlainTextComponentSerializer.plainText() .serialize(Component.translatable(materialParts[0]));
+			BooleanOreSetting setting = new BooleanOreSetting(plugin, true,
+					"Safe " + materialName + " break",
+					"safeOreBreak_" + materialParts[0].getKey().getKey(),
+					"Prevents you from breaking " + materialName
+							+ " without a silk touch pickaxe.",
+					materialParts[0]);
+			oreSettings.add(setting);
+			PlayerSettingAPI.registerSetting(setting, mainMenu);
+		}
+	}
+
+	@Override
+	public void onDisable() {
+		super.onDisable();
+	}
+
+	@EventHandler(ignoreCancelled = true)
+	public void onOreBreak(BlockBreakEvent event) {
+		if (event.getPlayer().getGameMode() != GameMode.SURVIVAL) {
+			return;
+		}
+
+		ItemStack item = event.getPlayer().getInventory().getItemInMainHand();
+		if (item.hasItemMeta() && item.getItemMeta().hasEnchant(Enchantment.SILK_TOUCH)) {
+			return;
+		}
+
+		OUTER:
+		for (int i = 0; i < breakOres.size(); i++) {
+			List<Material> list = breakOres.get(i);
+			for (Material material : list) {
+				if (material == event.getBlock().getType()) {
+					Boolean value = oreSettings.get(i).getValue(event.getPlayer());
+
+					if (!value) {
+						continue OUTER;
+					}
+
+					if (receivedWarning.add(event.getPlayer())) {
+						event.getPlayer().sendMessage(Component.text(
+										"A SimpleAdminHacks /config option is preventing you from breaking that ore without a silk touch pickaxe.")
+								.color(NamedTextColor.RED));
+					}
+					event.setCancelled(true);
+					return;
+				}
+			}
+		}
+	}
+
+	private static class BooleanOreSetting extends PlayerSetting<Boolean> {
+
+		private final Material ore;
```

Comment:
You should refrain from caching the value this way since it invalidates the [`/config set <player|uuid> <setting> <value>`](https://github.com/CivMC/CivModCore/blob/321ab959cfa72546551eb2600db6fa56e6bc742f/paper/src/main/java/vg/civcraft/mc/civmodcore/players/settings/commands/ConfigCommand.java#L60-L85) admin command. Or at least override the respective methods to also update your cache-field here.

Reply:
![image](https://user-images.githubusercontent.com/52572989/226786609-e6d7e8fb-2602-4c85-bee1-e0beca72e8d1.png)
You know what I mean.